library/src/types.ts

```diff
@@ -138,17 +172,72 @@ export type PipeResult<TOutput> =
       issues: Pick<Issue, 'validation' | 'message' | 'input' | 'path'>[];
     };
 
+export type Validation<
+  TValue,
+  TKind extends string = string,
+  TMessage extends ErrorMessage = ErrorMessage,
+  TRequirement = unknown
+> = {
+  /* The validation type */
+  kind: TKind;
+  /* The validation error message */
+  message: TMessage;
+  /* The requirement value to compare against (Optional) */
+  requirement?: TRequirement;
```

Comment:
I an optional `requirement` field really a good idea? If I implement a validation function that always sets the `requirement` field it is somehow awkward to always add a guard just because the type is not precise enough. If someone wants to have a validation function without a `requirement` field it might be better to explicitly set it to `undefined` or `null`. WDYT?